todo_app/data/todoapi.py

```diff
@@ -0,0 +1,160 @@
+from abc import ABC, abstractmethod
+from configparser import ConfigParser
+from .boardelements import *
+import os
+import requests
+
+_TRELLO = 'trello'
+
+def todo_api_factory(api_name):
+    ''' Factory method for creating a concrete ToDoAPI. Currently only accepts 'trello' '''
+    if api_name == _TRELLO:
+        return TrelloAPI()
+    else:
+        #TODO add more APIs in the future
+        return None
+
+
+class ToDoAPI(ABC):
```

Comment:
I'd argue this is a case of YAGNI, but hey you never know....

Reply:
Typically you'd have this in a separate file (ideally you'd have 1 file per class)

Reply:
Agree it is YAGNI. Removed